### PR TITLE
Test docker agent

### PR DESCRIPTION
**Pull request by luigibk**
Resolves # . (This is applicable only if this pull request relates to a GitHub issue, delete the line otherwise)

[AB#901]

Notes:
*
*
*

### CODE REVIEW

File: vars/withAcrClient.groovy [hunk 1]
```diff
@@ -1,11 +1,11 @@
 def call(String subscription, Closure block) {
-  withDocker('hmcts/cnp-aks-client:az-2.3.0-kubectl-1.18.0-helm-3.1.2', null) {
+//  withDocker('hmcts/cnp-aks-client:az-2.3.0-kubectl-1.18.0-helm-3.1.2', null) {
```
Comment (timja) [inline vars/withAcrClient.groovy:2]: so we need to add these to the VM agent image as well then?

Comment (luigibk) [reply vars/withAcrClient.groovy:2]: I'd rather not if possible...

File: vars/withPipeline.groovy [hunk 2]
```diff
@@ -59,9 +60,23 @@ def call(type, String product, String component, Closure body) {
...
+
+  node(agentType) {
+    def slackChannel = teamConfig.getBuildNoticesSlackChannel(product)
     try {
+      if (teamConfig.isDockerBuildAgent(product)) {
+        def envName = env.PROD_ENVIRONMENT_NAME
+        withSubscriptionLogin(envName) {
+          def homeDir = "/home/jenkins"
+          def mkdirOut = sh(label: "mkdir .ssh", script: "[ ! -d '${homeDir}/.ssh' ] && mkdir -p '${homeDir}/.ssh' && chmod 700 '${homeDir}/.ssh' || exit 0", returnStdout: true)?.trim()
+          echo("'mkdir .ssh' output: ${mkdirOut}")
```
Comment (timja) [inline vars/withPipeline.groovy:74]:
```suggestion
```

File: vars/withPipeline.groovy [hunk 3]
```diff
@@ -59,9 +60,23 @@ def call(type, String product, String component, Closure body) {
...
+          def infraVaultName = envName == "sandbox" ? "infra-vault-sandbox" : "infra-vault-prod"
```
Comment (timja) [inline vars/withPipeline.groovy:75]:
```suggestion
          def infraVaultName = env.INFRA_VAULT_NAME
```

**Pull request merged**